File: src/sdk-reference/cpp/1/realtime/subscribe/snippets/document-notifications.cpp
kuzzleio::NotificationListener listener =
  [](const kuzzleio::notification_result *notification) {
    std::string id = notification->result->id;

    if (notification->scope == std::string("in")) {
      std::cout << "Document " << id << " enter the scope" << std::endl;
    } else {
      std::cout << "Document " << id << " leave the scope" << std::endl;
    }
  };

try {
  // Subscribe to notifications for documents containing a 'name' property
  std::string filters = R"({ "exists": "name" })";
  kuzzle->realtime->subscribe("nyc-open-data", "yellow-taxi", filters, &listener);

  // Create a document matching the provided filters
  kuzzle->document->create(
    "nyc-open-data",
    "yellow-taxi",
    "nina-vkote",
    R"({ "name": "nina vkote", "age": 19 })");
} catch (kuzzleio::KuzzleException &e) {
  std::cerr << e.what() << std::endl;
}

File: src/sdk-reference/csharp/1/realtime/subscribe/snippets/document-notifications.cs
NotificationListener listener =
  [](const notification_result *notification) {
    string id = notification.result.id;

    if (notification.scope == string("in")) {
      Console.WriteLine("Document " + id + " enter the scope");
    } else {
      Console.WriteLine("Document " + id + " leave the scope");
    }
  };

try {
  // Subscribe to notifications for documents containing a 'name' property
  string filters = @"{ ""exists"": ""name"" }";
  kuzzle.realtime.subscribe("nyc-open-data", "yellow-taxi", filters, listener);

  // Create a document matching the provided filters
  kuzzle.document.create(
    "nyc-open-data",
    "yellow-taxi",
    "nina-vkote",
    @"{ ""name"": ""nina vkote"", ""age"": 19 }");
} catch (KuzzleException e) {
  Console.Error.WriteLine(e.what());
}
